reduction/lr_reduction/DeadTimeCorrection.py

```diff
@@ -0,0 +1,98 @@
+"""
+    Dead time correction algorithm for single-readout detectors.
+"""
+import time
+import math
+import os
+from mantid.api import *
+from mantid.simpleapi import *
+from mantid.kernel import *
+import numpy as np
+import scipy
+
+def call(InputWorkspace, DeadTime=4.2, TOFStep=100, Paralyzable=False, TOFRange=[0, 0], OutputWorkspace='correction'):
+    """
+        Function to make the algorithm call similar to a normal Mantid call
+    """
+    algo = SingleReadoutDeadTimeCorrection()
+    algo.PyInit()
+    algo.setProperty("InputWorkspace", InputWorkspace)
+    algo.setProperty("DeadTime", DeadTime)
+    algo.setProperty("TOFStep", TOFStep)
+    algo.setProperty("Paralyzable", Paralyzable)
+    algo.setProperty("TOFRange", TOFRange)
+    algo.setProperty("OutputWorkspace", OutputWorkspace)
+    algo.PyExec()
+    return algo.getProperty('OutputWorkspace').value
+
+
+class SingleReadoutDeadTimeCorrection(PythonAlgorithm):
+
+    def category(self):
+        return "Reflectometry\\SNS"
+
+    def name(self):
+        return "SingleReadoutDeadTimeCorrection"
+
+    def version(self):
+        return 1
+
+    def summary(self):
+        return "Single read-out dead time correction calculation"
+
+    def PyInit(self):
+        self.declareProperty(WorkspaceProperty("InputWorkspace", "", Direction.Input),
+                             "Optionally, we can provide a workspace directly")
```

Comment:
I don't understand this tooltip. is this `InputWorkspace` a required property? Why "optionally"?